### DISCUSSION TITLE
Custom upload button

### DISCUSSION THREAD

**Discussion by Varsha-R** [uploadcare/file-uploader#395] [blocks] [answered]
Hi! 
I have been trying to add the blocks modal to an existing menu button that we have in our application, but it only works with `<lr-file-uploader-regular>`, `<ldr-simple-btn>`, and `<ldr-source-btn>`.
Is there a way to customize this 'Upload files' button? (I want to be able to use the same on-click functionality but with my custom button)

**Comment by nd0ut** [comment, marked as answer]
Hey @Varsha-R, yep, that's possible! 

Here is an example: https://codesandbox.io/s/4plsdv?file=/index.js

`lr-upload-ctx-provider` block exposes public context API and it's undocumented yet

**Reply by barbalex** [reply]
Does this still work with v0.32.4? I am getting undefined instead.
My fork of your sandbox does not work when using 0.32.4: https://codesandbox.io/p/sandbox/reverent-ishizaka-8t73w8?file=%2Findex.html%3A11%2C70
